src/hotspot/cpu/s390/s390.ad

```diff
@@ -10688,44 +10732,30 @@ instruct popCountI(iRegI dst, iRegI src, iRegI tmp, flagsReg cr) %{
     Register Rtmp = $tmp$$Register;
 
     // Prefer compile-time assertion over run-time SIGILL.
-    assert(VM_Version::has_PopCount(), "bad predicate for countLeadingZerosI");
-    assert_different_registers(Rdst, Rtmp);
+    assert(VM_Version::has_PopCount(), "bad predicate for popCountI");
```

Comment:
Duplicate check! But you may want to check for different registers (if you don't trust the effect().